### ISSUE TITLE
`v-bind` shorthand syntax without argument (`:=`) generates `vue/no-unused-vars` false positive

### ISSUE THREAD

**Issue by Neluji** [vuejs/eslint-plugin-vue#2462]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.57.0
- **eslint-plugin-vue version:** 9.26.0
- **Vue version:** 3.4.27
- **Node version:** 18.20.0
- **Operating System:** Linux (Ubuntu 23.04)

**Please show your full configuration:**

```json5
/* eslint-env node */
require('@rushstack/eslint-patch/modern-module-resolution')

module.exports = {
  root: true,
  'extends': [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    '@vue/eslint-config-typescript',
    '@vue/eslint-config-prettier/skip-formatting'
  ],
  parserOptions: {
    ecmaVersion: 'latest'
  }
}

```

**What did you do?**	

```vue
<!-- MyComponent.vue -->

<template>
  <slot :props="{ id: 'mySlotId' }"></slot>
</template>

<!-- App.vue -->

<script setup lang="ts">
import { ref } from 'vue'
import MyComponent from './components/MyComponent.vue'

const id = ref({ id: 'myId' })
</script>

<template>
  <div :="id" />

  <MyComponent>
    <template #default="{ props }">
      <div :="props" />
    </template>
  </MyComponent>

  <MyComponent #="{ props }">
    <div :="props" />
  </MyComponent>
</template>
```

**What did you expect to happen?**

According to [Vue Style Guide - Priority B rules: _Directive Shorthands_](https://vuejs.org/style-guide/rules-strongly-recommended.html#directive-shorthands), the use of directive shorthands should be consistent across the application code.

From this, I understand that when using them, `v-bind` should never be used, but rather its shorthand `:` instead, **including when using it [without arguments to bind multiple attributes](https://vuejs.org/guide/essentials/template-syntax.html#dynamically-binding-multiple-attributes)** ; i.e. `:="myAttribute"` should be used instead of `v-bind="myAttribute`.

Thus, this usage should **not give any error or warning**.

**What actually happened?**

The **`vue/no-unused-vars` rule generates an error** when using **this syntax in named scoped slots with slot props**. Note that it is not the case when using default slot props without using the `<template>` tag (i.e. directly from the component tag, see example).

<details>

<summary><em>ESLint output</em></summary>

```sh

> vue-project@0.0.0 lint /tmp/vue-project
> eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore


/tmp/vue-project/src/App.vue
   7:7   warning  'id' is assigned a value but never used  @typescript-eslint/no-unused-vars
  14:27  error    'props' is defined but never used        vue/no-unused-vars

✖ 2 problems (1 error, 1 warning)

 ELIFECYCLE  Command failed with exit code 1.
```

</details>

**Repository to reproduce this issue**

https://github.com/Neluji/issue-examples/tree/vuejs/eslint-plugin-vue
(be sure to be on branch `vuejs/eslint-plugin-vue`)

**Comment by jacekkarczmarczyk** [comment]
https://github.com/vuejs/eslint-plugin-vue/issues/2206
